test/model/typing_status_test.dart

```diff
@@ -0,0 +1,207 @@
+import 'package:checks/checks.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:zulip/api/model/events.dart';
+import 'package:zulip/api/model/model.dart';
+import 'package:zulip/model/narrow.dart';
+import 'package:zulip/model/typing_status.dart';
+
+import '../example_data.dart' as eg;
+import '../fake_async.dart';
+
+void main() {
+  late TypingStatus model;
+  late int notifiedCount;
+
+  void prepareModel() {
+    model = TypingStatus(
+      selfUserId: eg.selfUser.userId,
+      typingStartedExpiryPeriod: const Duration(milliseconds: 15000));
+    check(model.debugActiveNarrows).isEmpty();
+    notifiedCount = 0;
+    model.addListener(() => notifiedCount += 1);
+  }
+
+  void checkNotNotified() {
+    check(notifiedCount).equals(0);
+  }
+
+  void checkNotifiedOnce() {
+    check(notifiedCount).equals(1);
+    notifiedCount = 0;
+  }
+
+  void checkTypists(Map<SendableNarrow, List<User>> typistsByNarrow) {
+    final actualTypistsByNarrow = <SendableNarrow, Iterable<int>>{};
+    for (final narrow in model.debugActiveNarrows) {
+      actualTypistsByNarrow[narrow] = model.typistIdsInNarrow(narrow);
+    }
+    check(actualTypistsByNarrow).deepEquals(
+      typistsByNarrow.map((k, v) => MapEntry(k, v.map((e) => e.userId))));
+  }
+
+  final stream = eg.stream();
+  final topicNarrow = TopicNarrow(stream.streamId, 'foo');
+
+  final dmNarrow = DmNarrow.withUser(eg.otherUser.userId, selfUserId: eg.selfUser.userId);
+  final groupNarrow = DmNarrow.withOtherUsers(
+    [eg.otherUser.userId, eg.thirdUser.userId], selfUserId: eg.selfUser.userId);
+
+  group('handle typing start events', () {
+    test('add typists in separate narrows', () {
+      prepareModel();
+
+      model.handleTypingEvent(eg.typingEvent(dmNarrow, TypingOp.start, eg.otherUser.userId));
```

Comment:
A lot of these `model.handleTypingEvent` lines have key information — the identity of the typist — beyond column 80.

So let's shorten with a little helper, as I alluded to at the end of https://github.com/zulip/zulip-flutter/pull/783#discussion_r1671621238 . Can look like:
```suggestion
      handleTypingEvent(dmNarrow, TypingOp.start, eg.otherUser);
```